festim/boundary_conditions/fluxes/dissociation_flux.py

```diff
@@ -0,0 +1,32 @@
+from festim import FluxBC, k_B
+import fenics as f
+import sympy as sp
+
+
+class DissociationFlux(FluxBC):
+    """
+    FluxBC subclass for hydrogen dissociation flux.
+    -D(T) * grad(c) * n = -Kd(T) * P
+
+    Args:
+        Kd_0 (float or sp.Expr): dissociation coefficient pre-exponential
+            factor (m^(2)/s)
+        E_Kd (float or sp.Expr): dissociation coefficient activation
+            energy (eV)
+        P (float): partial pressure of H (Pa)
+        surfaces (list or int): the surfaces of the BC
+    """
+
+    def __init__(self, Kd_0, E_Kd, P, surfaces) -> None:
+        self.Kd_0 = Kd_0
+        self.E_Kd = E_Kd
+        self.P = P
+        super().__init__(surfaces=surfaces, field=0)
+
+    def create_form(self, T):
+        Kd_0_expr = f.Expression(sp.printing.ccode(self.Kd_0), t=0, degree=1)
+        E_Kd_expr = f.Expression(sp.printing.ccode(self.E_Kd), t=0, degree=1)
```

Comment:
Note:

having Expression objects for everything comes at a cost. these objects are more costly than Constant() objects for instance.
Especially since we keep "updating" them at each timestep even though they don't evolve with time.
In the future, it'd be nice to have some kind of tests to check: is it a constant? is it only a function of space? is it only a function of time? and use the most appropriate fenics object then.

Does that make sense?

Reply:
Yeah I get what you mean, but would we need to make this test in every class that creates a form?

Reply:
yes